app/subscribers/publish_zync_event_subscriber.rb

```diff
@@ -10,25 +10,42 @@ def initialize(publisher = DEFAULT_PUBLISHER)
 
   attr_reader :publisher
 
-  # @param [ZyncEvent] event
-  def call(event)
-    unless ThreeScale.config.onpremises
+  delegate :domain_event_in_saas?, :build_zync_event, to: :class
+
+  class << self
+    def domain_event_in_saas?(event)
+      ThreeScale.saas? && (event.is_a?(Domains::ProxyDomainsChangedEvent) || event.is_a?(Domains::ProviderDomainsChangedEvent))
+    end
+
+    def build_zync_event(event)
       case event
-      when Domains::ProxyDomainsChangedEvent, Domains::ProviderDomainsChangedEvent
-        return
+      when ApplicationRelatedEvent
+        event.service&.oauth? || sync_non_oidc_apps ? ZyncEvent.create(event, event.application) : nil
```

Comment:
One question that popped now in my mind. DO we observe sometimes `event.service` to be `nil`?

Reply:
Mmm. I don't know if that can still happen, to be honest. I think it shouldn't... but you never know :sweat_smile: 

Is it that `&.` bothers you?